src/panels/lovelace/components/hui-actions-editor.ts

```diff
@@ -0,0 +1,233 @@
+import "@material/mwc-icon-button";
+import {
+  mdiClose,
+  mdiGestureDoubleTap,
+  mdiGestureTap,
+  mdiGestureTapHold,
+  mdiPencil,
+} from "@mdi/js";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+
+import { fireEvent } from "../../../common/dom/fire_event";
+import { ActionConfig } from "../../../data/lovelace";
+import { HomeAssistant } from "../../../types";
+
+import "../../../components/ha-svg-icon";
+
+export interface EditActionEvent {
+  type: string;
+}
+
+declare global {
+  interface HASSDomEvents {
+    "edit-action": EditActionEvent;
+    "clear-action": EditActionEvent;
+  }
+}
+
+@customElement("hui-actions-editor")
+export class HuiActionsEditor extends LitElement {
+  @property({ attribute: false }) protected hass?: HomeAssistant;
+
+  @property({ attribute: false }) protected tapAction?: ActionConfig;
+
+  @property({ attribute: false }) protected holdAction?: ActionConfig;
+
+  @property({ attribute: false }) protected doubleTapAction?: ActionConfig;
+
+  protected render(): TemplateResult {
+    if (!this.hass) {
+      return html``;
+    }
+
+    return html`
+      <h3>
+        ${this.hass!.localize("ui.panel.lovelace.editor.card.generic.actions")}
+      </h3>
+      <div class="actions">
+        ${this.tapAction
+          ? html` <div class="action">
+              <div>
+                <ha-svg-icon .path=${mdiGestureTap}></ha-svg-icon>
+                <span
+                  >${this.hass!.localize(
+                    "ui.panel.lovelace.editor.card.generic.tap_action"
+                  )}
+                  -
+                  ${this.hass!.localize(
+                    `ui.panel.lovelace.editor.action-editor.actions.${this.tapAction?.action}`
+                  ) ||
+                  this.hass!.localize(
+                    "ui.panel.lovelace.editor.action-editor.actions.none"
+                  )}</span
+                >
+              </div>
+              <div>
+                <mwc-icon-button
+                  aria-label=${this.hass!.localize(
+                    "ui.components.entity.entity-picker.edit"
+                  )}
+                  class="edit-icon"
+                  .type=${"tap_action"}
+                  @click=${this._editAction}
+                >
+                  <ha-svg-icon
+                    .path=${mdiPencil}
+                    .type=${"tap_action"}
+                  ></ha-svg-icon>
+                </mwc-icon-button>
+                <mwc-icon-button
+                  aria-label=${this.hass!.localize(
+                    "ui.components.entity.entity-picker.clear"
+                  )}
+                  class="remove-icon"
+                  .type=${"tap_action"}
+                  @click=${this._clearAction}
+                >
+                  <ha-svg-icon
+                    .path=${mdiClose}
+                    .type=${"tap_action"}
```

Comment:
Why the type of both the icon and the button?

Reply:
I found that if I clicked on the very edge of the button, it would just be the mwc-icon-button, but if I clicked towards the center more it was the ha-svg-icon. Is there a better solution for that?

Reply:
yes, use `currentTarget` instead of `target` in the event handler.